### ISSUE TITLE
knife subcommand --help doesn't work as intended

### ISSUE THREAD

**Issue by moutons** [chef/chef#8848]
:evilchef:
```
# chef --version
Chef Development Kit Version: 1.3.43
...
# chef gem install knife-acl
Fetching: knife-acl-1.0.3.gem (100%)
Successfully installed knife-acl-1.0.3
1 gem installed
# knife group --help
FATAL: Cannot find subcommand for: 'group --help'
Available subcommands: (for details, knife SUB-COMMAND --help)
...
```

**Comment by Vasu1105** [comment]
Latest chefdk `4.3.13` don't gives above error but it does not show the expected result. It should filter out only knife group commands from list, but it lists all the knife command when we run ```knife group --help ``` also same is the result when we run ```knife acl --help```

**Comment by Vasu1105** [comment]
My bad it does give above error and then list the command

**Comment by Vasu1105** [comment]
Fixed in #8915
